pynamodb/models.py

```diff
@@ -267,6 +264,18 @@ def __init__(self, name, bases, namespace, discriminator=None) -> None:
                 }
                 cls.DoesNotExist = type('DoesNotExist', (DoesNotExist, ), exception_attrs)
 
+    @staticmethod
+    def _initialize_indexes(cls):
+        """
+        Initialize indexes on the class.
+        """
+        cls._indexes = {}
+        for name, index in getmembers(cls, lambda o: isinstance(o, Index)):
+            index.Meta.model = cls
```

Comment:
This changes the current behavior where the class is set once by the parent model and not overwritten by child classes. The new (still broken) behavior is last-write-wins during child model creation.